Dockerfile.ubi.cpu.ppc64le

```
@@ -0,0 +1,247 @@
+    source /opt/rh/gcc-toolset-13/enable && \
+    pip install -v \
+        /arrowwheels/*.whl \
+        /opencvwheels/*.whl \
+        /torchwheels/*.whl && \
+    cd /vllm && \
+    pip install pandas==2.0 && \
+    pip install -v ".[tensorizer]" --no-build-isolation && \
+    python -m build -w -o /wheelsdir/ --no-isolation && \
+    cd .. && \
+    git clone https://github.com/opendatahub-io/vllm-tgis-adapter.git -b ${VLLM_TGIS_ADAPTER_VERSION} && \
+    cd vllm-tgis-adapter && \
+    git cherry-pick --no-commit ${TGIS_PATCH} && \
+    python -m build -w -o /wheelsdir/
+
+
+###############################################################
+#                   FINAL VLLM IMAGE STAGE                    #
+###############################################################
+
+FROM registry.access.redhat.com/ubi9/ubi-minimal
+
+ARG PYTHON_VERSION=3.12
+ARG OPENBLAS_VERSION=0.3.29
+
+# Set Environment Variables for venv & openblas
+ENV PATH=/opt/vllm/bin:$PATH
+ENV PKG_CONFIG_PATH=/usr/local/lib/pkgconfig/
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib64:/usr/local/lib:/usr/lib64:/usr/lib
+
+
+# create artificial dependencies between stages for independent stages to build in parallel
+COPY --from=torch-builder /tmp/control /dev/null
+COPY --from=arrow-builder /tmp/control /dev/null
+COPY --from=cv-builder /tmp/control /dev/null
+COPY --from=vllm-deps /tmp/control /dev/null
+
+# install gcc-13, python, openblas
+RUN rpm -ivh https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm && \
+    microdnf install --nodocs -y \
+    tar findutils unzip openssl numactl lapack \
+    pkgconfig xsimd gcc-toolset-13 libsndfile \
+    libtiff libjpeg openjpeg2 zlib zeromq \
+    freetype lcms2 libwebp tcl tk utf8proc \
+    harfbuzz fribidi libraqm libimagequant libxcb \
+    python${PYTHON_VERSION}-devel python${PYTHON_VERSION}-pip \
+    && microdnf clean all \
+    && python${PYTHON_VERSION} -m venv /opt/vllm \
+    && python -m pip install -U pip uv \
+    && curl -sL https://ftp2.osuosl.org/pub/ppc64el/openblas/latest/Openblas_${OPENBLAS_VERSION}_ppc64le.tar.gz | tar xvf - -C /usr/local
+
+# use pip instead of uv to reuse the pip cache from vllm-deps stage
+# mount cache directory from vllm-deps stage to avoid redundant compilation
+RUN --mount=type=cache,target=/root/.cache/pip \
+    --mount=type=cache,target=/root/.cache/uv \
+    --mount=type=bind,from=torch-builder,source=/torchwheels/,target=/torchwheels/,ro \
+    --mount=type=bind,from=arrow-builder,source=/arrowwheels/,target=/arrowwheels/,ro \
+    --mount=type=bind,from=cv-builder,source=/opencvwheels/,target=/opencvwheels/,ro \
+    --mount=type=bind,from=vllm-deps,source=/wheelsdir/,target=/wheelsdir/,ro \
+    --mount=type=bind,from=vllm-deps,source=/root/.cache,target=/root/.cache,ro \
+    pip install pandas==2.0 && \
+    pip install -v \
+        /arrowwheels/*.whl \
+        /opencvwheels/*.whl \
+        /torchwheels/*.whl \
+        /wheelsdir/*.whl
```

```suggestion
RUN --mount=type=cache,target=/root/.cache/uv \
    --mount=type=bind,from=torch-builder,source=/torchwheels/,target=/torchwheels/,ro \
    --mount=type=bind,from=arrow-builder,source=/arrowwheels/,target=/arrowwheels/,ro \
    --mount=type=bind,from=cv-builder,source=/opencvwheels/,target=/opencvwheels/,ro \
    --mount=type=bind,from=vllm-deps,source=/wheelsdir/,target=/wheelsdir/,ro \
    uv pip install pandas==2.0 && \
    uv pip install -v \
        /arrowwheels/*.whl \
        /opencvwheels/*.whl \
        /torchwheels/*.whl \
        /wheelsdir/*.whl
```

- the main point here is to avoid mixing `pip` and `uv` usage and to make sure that we're only using a single cache
- make sure that we use `--installer=uv` everywhere when using `python -m build` (without isolation)
- if we mount the caches correctly and use `uv` everywhere, there should be no need to bind mount `/root/.cache`
- we need to add `vllm-tgis-adapter` here if not building it in the previous step (we shouldn't need to build it since it's a pure python wheel)
- do we need to install `pandas` in a separate step? Do we really need to pin it to 2.0?